src/components/ui/toolbox.ts

```diff
@@ -346,9 +393,15 @@ export default class Toolbox extends EventsDispatcher<ToolboxEvent> {
      */
     const index = currentBlock.isEmpty ? currentBlockIndex : currentBlockIndex + 1;
 
+    /**
+     * Merge real tool's data with data overrides
+     */
+    const defaultBlockData = await this.api.blocks.composeBlockData(toolName);
```

Comment:
I think we should do that only if we have the `blockDataOverrides`. In other cases (insertion of a regular tool) we will skip extra block construction/saving/merging.